cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -638,23 +668,33 @@ private SubmitJobRequest getSubmitJobRequest(RuntimeJobInfo runtimeJobInfo, List
       }
     }
 
+    Job.Builder dataprocJobBuilder = Job.newBuilder()
+      // use program run uuid as hadoop job id on dataproc
+      .setReference(JobReference.newBuilder().setJobId(getJobId(runInfo)))
+      // place the job on provisioned cluster
+      .setPlacement(JobPlacement.newBuilder().setClusterName(clusterName).build())
+      // add same labels as provisioned cluster
+      .putAllLabels(labels)
+      // Job label values must match the pattern '[\p{Ll}\p{Lo}\p{N}_-]{0,63}'
+      // Since program name and type are class names they should follow that pattern once we remove all
+      // capitals
+      .putLabels(LABEL_CDAP_PROGRAM, runInfo.getProgram().toLowerCase())
+      .putLabels(LABEL_CDAP_PROGRAM_TYPE, runInfo.getProgramType().toLowerCase())
+      .setHadoopJob(hadoopJobBuilder.build());
+
+    GetClusterRequest getClusterRequest = GetClusterRequest.newBuilder()
+      .setClusterName(clusterName).setProjectId(projectId).setRegion(region).build();
+
+    // if the dataproc cluster has driver pools enabled add driver scheduling config
+    if (getClusterControllerClient().getCluster(getClusterRequest).getConfig().getAuxiliaryNodeGroupsCount() > 0) {
```

Comment:
we may hit cluster operations quota limit. Is this check necessary for every launch specially for existing clusters?

Reply:
we might need to add a config to avoid getCluster calls under high load.

Reply:
added a config to skip the check.

Reply:
Discussed offline, as the other quotas like `GetJobRequestsPerMinutePerProjectPerRegion` is counted independently from `RequestsPerMinutePerProjectPerRegion` we can keep this check.

Reply:
Great! But please add this context as a comment in code.

Reply:
done